### ISSUE TITLE
Pulsar trigger failing with invalid Admin URL showing %!s(<nil>)

### ISSUE THREAD

**Issue by gabron** [kedacore/keda#4341]
### Report

We run a solution that creates namespaces and deploys applications to it dynamically.  We deploy pulsar in a standalone mode into the namespace, with the services that connect to it as well as the scaled objects.  Configuring the keda scaler, with no authentication resembles this trigger configuration.

```
  triggers:
    - type: pulsar
      metadata:
        adminURL: http://pulsar.job-run-234701ce-57a3-4668-98bf-306b75fdb909:8080
        topic: persistent://public/default/nlp-verbnet
        isPartitionedTopic: "false"
        subscription: nlp-verbnet
        msgBacklogThreshold: '100'
        activationMsgBacklogThreshold: '0'
```

The scaledobjects get created, the corresponding HPA objects do as well.  The scaledobjects initially post as ready and then revert back to not ready or False.  No deployment ever gets created even though the `minReplicaCount: 1` is set.

### Expected Behavior

I would expect the in cluster pulsar to be able to be referenced by the scaler trigger as a source of scaling, and the scaledobjects to show as ready and healthy.

### Actual Behavior

Scaledobjects get created and corresponding pod autoscaler objects but no pods ever spin up.

### Steps to Reproduce the Problem

1. Deploy pulsar to a namespace in standalone mode
2. Define and deploy a service that will consume from pulsar into the same namespace
3. Define and deploy a scaledobject that will use a pulsar trigger and link it to the adminURL of the pulsar service <servicename.namespace>.  Set minReplicaCount to 1

### Logs from KEDA operator

```
2023-03-08T18:27:36Z	DEBUG	scaleexecutor	Triggers defined in ScaledObject are not working correctly	{"scaledobject.Name": "verbnet", "scaledObject.Namespace": "job-run-234701ce-57a3-4668-98bf-306b75fdb909", "scaleTarget.Name": "nlp-verbnet"}
2023-03-08T18:27:36Z	DEBUG	events	error requesting stats from url: error requesting stats from url: %!s(<nil>)	{"type": "Warning", "object": {"kind":"ScaledObject","namespace":"job-run-234701ce-57a3-4668-98bf-306b75fdb909","name":"verbnet","uid":"cfe28f0a-3129-4274-bf41-2260fa927466","apiVersion":"keda.sh/v1alpha1","resourceVersion":"973063"}, "reason": "KEDAScalerFailed"}
2023-03-08T18:27:58Z	ERROR	scalers_cache	error getting scale decision	{"scaledobject.Name": "deparse", "scaledObject.Namespace": "job-run-234701ce-57a3-4668-98bf-306b75fdb909", "scaleTarget.Name": "nlp-deparse", "error": "error requesting stats from url: error requesting stats from url: %!s(<nil>)"}
github.com/kedacore/keda/v2/pkg/scaling/cache.(*ScalersCache).GetScaledObjectState
	/workspace/pkg/scaling/cache/scalers_cache.go:155
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers
	/workspace/pkg/scaling/scale_handler.go:360
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop
	/workspace/pkg/scaling/scale_handler.go:162
```

### KEDA Version

2.9.3

### Kubernetes Version

1.24

### Platform

Google Cloud

### Scaler Details

Pulsar

### Anything else?

If I remove the namespace from the adminURL the keda operator will print out the full path it is trying to resolve.

```
error": "error requesting stats from url: error requesting stats from url: Get \"http://pulsar:8080/admin/v2/persistent/public/default/nlp-verbnet/stats\": dial tcp: lookup pulsar on 10.0.16.10:53: no such host"
```
This clearly isn't resolvable by kube dns outside of the namespace, but shows something other than `nil`

We are running pulsar version 2.10.3

**Comment by gabron** [comment]
This could be a configuration issue on my side, I'm failing to find a good explanation online as to why it would be, I do see this error over and over again in the keda metrics server logs

```
W0309 17:26:51.591977       1 logging.go:59] [core] [Channel #1 SubChannel #2] grpc: addrConn.createTransport failed to connect to {
  "Addr": "keda-operator.keda.svc.cluster.local:9666",
  "ServerName": "keda-operator.keda.svc.cluster.local:9666",
  "Attributes": null,
  "BalancerAttributes": null,
  "Type": 0,
  "Metadata": null
}. Err: connection error: desc = "transport: Error while dialing dial tcp: lookup keda-operator.keda.svc.cluster.local on 169.254.169.254:53: no such host"
I0309 17:27:51.524673       1 client.go:85] keda_metrics_adapter/provider "msg"="Waiting for establishing a gRPC connection to KEDA Metrics Server" 
E0309 17:27:51.580963       1 provider.go:96] keda_metrics_adapter/provider "msg"="timeout" "error"="timeout while waiting to establish gRPC connection to KEDA Metrics Server" 
E0309 17:27:51.581097       1 writers.go:118] apiserver was unable to write a JSON response: expected pointer, but got nil
E0309 17:27:51.581191       1 status.go:71] apiserver received an error that is not an metav1.Status: &errors.errorString{s:"expected pointer, but got nil"}: expected pointer, but got nil
E0309 17:27:51.582941       1 writers.go:131] apiserver was unable to write a fallback JSON response: http: Handler timeout
I0309 17:27:51.584108       1 trace.go:219] Trace[1184124260]: "List" url:/apis/external.metrics.k8s.io/v1beta1/namespaces/job-run-e3032d20-3f2d-4246-9628-fbe4415590b9/s0-pulsar-persistent---public-default-nlp-verbnet-nlp-verbnet,user-agent:vpa-recommender/v0.0.0 (linux/amd64) kubernetes/$Format/metrics-horizontal-pod-autoscaler,audit-id:75ec425e-fb2f-4f6b-8851-d9befc2fd39a,client:172.16.0.4,accept:application/vnd.kubernetes.protobuf, */*,protocol:HTTP/2.0 (09-Mar-2023 17:26:51.514) (total time: 60069ms):
Trace[1184124260]: ---"Listing from storage done" 60066ms (17:27:51.580)
Trace[1184124260]: [1m0.069412713s] [1m0.069412713s] END
E0309 17:27:51.584646       1 timeout.go:141] post-timeout activity - time-elapsed: 79.392024ms, GET "/apis/external.metrics.k8s.io/v1beta1/namespaces/job-run-e3032d20-3f2d-4246-9628-fbe4415590b9/s0-pulsar-persistent---public-default-nlp-verbnet-nlp-verbnet" result: <nil>
```

**Comment by gabron** [comment]
Turns out that the issue had to do with our architecture.  We created subscriptions dynamically.  This being the case the subscription wouldn't exist when we deployed the scaledobject and as a result the URL for getting stats wouldn't exist.  We set a fallback on each scaledobject to create a minimum of one replica, after the replica was created and the subscription existed the URL for stats would then resolve and the scaledobject would work.
